src/ExpressionEvaluator/Core/Source/ExpressionCompiler/DynamicFlagsCustomTypeInfo_Factory.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.ExpressionEvaluator
+{
+    internal partial struct DynamicFlagsCustomTypeInfo
+    {
+        public static DynamicFlagsCustomTypeInfo Create(CustomTypeInfo customTypeInfo)
+        {
+            return new DynamicFlagsCustomTypeInfo(customTypeInfo.PayloadTypeId == PayloadTypeId ? customTypeInfo.Payload : null);
+        }
+
+        public static DynamicFlagsCustomTypeInfo Create(ArrayBuilder<bool> dynamicFlags)
+        {
+            return new DynamicFlagsCustomTypeInfo(dynamicFlags, startIndex: 0);
+        }
+
+        public static DynamicFlagsCustomTypeInfo Create(ImmutableArray<bool> dynamicFlags)
+        {
+            if (dynamicFlags == null)
```

Comment:
At one point, I had a roslyn diagnostic for ImmutableArray null checks.  It might be time to resurrect it.  For the curious, there's no warning, since a lifted equality check exists.